test/intl402/Temporal/Calendar/prototype/yearMonthFromFields/reference-day.js

```diff
@@ -0,0 +1,133 @@
+TemporalHelpers.assertPlainYearMonth(
+  result3,
+  2021, 2, "M02",
+  "reference day is set correctly even if day is out of range (overflow reject)",
+  "ce", 2021, /* reference day = */ 1
+);
+
+const hebrew = new Temporal.Calendar("hebrew");
+
+const result4 = hebrew.yearMonthFromFields({ year: 5782, monthCode: "M04", day: 20 });
+TemporalHelpers.assertPlainYearMonth(
+  result4,
+  5782, 4, "M04",
+  "reference day is the first of the calendar month even if day is given",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 5
+);
+const isoFields = result4.getISOFields();
+assert.sameValue(isoFields.isoYear, 2021, "Tevet 5782 begins in ISO year 2021");
+assert.sameValue(isoFields.isoMonth, 12, "Tevet 5782 begins in ISO month 12");
+
+const result5 = hebrew.yearMonthFromFields({ year: 5783, monthCode: "M05L" }, { overflow: "constrain" });
+TemporalHelpers.assertPlainYearMonth(
+  result5,
+  5783, 5, "M05",
+  "month code M05L does not exist in year 5783 (overflow constrain)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 23
+);
+
+assert.throws(
+  RangeError,
+  () => hebrew.yearMonthFromFields({ year: 5783, monthCode: "M05L" }, { overflow: "reject" }),
+  "month code M05L does not exist in year 5783 (overflow reject)",
+);
+
+const result6 = hebrew.yearMonthFromFields({ year: 5783, month: 13 }, { overflow: "constrain" });
+TemporalHelpers.assertPlainYearMonth(
+  result6,
+  5783, 12, "M12",
+  "month 13 does not exist in year 5783 (overflow constrain)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 18
+);
+
+assert.throws(
+  RangeError,
+  () => hebrew.yearMonthFromFields({ year: 5783, month: 13 }, { overflow: "reject" }),
+  "month 13 does not exist in year 5783 (overflow reject)",
+);
+
+const result7 = hebrew.yearMonthFromFields({ year: 5782, monthCode: "M04", day: 50 }, { overflow: "constrain" });
+TemporalHelpers.assertPlainYearMonth(
+  result7,
+  5782, 4, "M04",
+  "reference day is set correctly even if day is out of range (overflow constrain)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 5
+);
+
+const result8 = hebrew.yearMonthFromFields({ year: 5782, monthCode: "M04", day: 50 }, { overflow: "reject" });
+TemporalHelpers.assertPlainYearMonth(
+  result8,
+  5782, 4, "M04",
+  "reference day is set correctly even if day is out of range (overflow reject)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 5
+);
+
+const chinese = new Temporal.Calendar("chinese");
+
+// Month codes, month indices, and the ISO reference days of the months in 2022
+const testData = [
+  // TODO: Sources conflict over whether M01L and M12L exist in _any_ year.
+  // Clarify this, and uncomment or delete as appropriate. ICU says no, and most
+  // implementations are backed by ICU, so for now we won't test for it.
+  // ["M01", 1, 1],
```

Comment:
Given that the ICU doc is over 10 years old and the ICU implementation disagrees with it, I'm inclined to trust the implementation and uncomment those lines.